apis/v1beta1/validation/gateway.go

```diff
@@ -157,3 +162,26 @@ func validateHostnameProtocolPort(listeners []gatewayv1b1.Listener, path *field.
 	}
 	return errs
 }
+
+// validateGatewayAddresses validates whether required fields of addresses are set according
+// to the Gateway API specification.
+func validateGatewayAddresses(addresses []gatewayv1b1.GatewayAddress, path *field.Path) field.ErrorList {
```

Comment:
can you also add a duplicate check in here

Reply:
great idea! added